orchestra/tests/test_todos.py

```diff
@@ -602,3 +632,241 @@ def test_conditional_skip_remove_todos_from_template(self):
         ]
         for todo, expected_todo in zip(todos, expected_todos):
             self._verify_todo_content(todo, expected_todo)
+
+
+class TodoListTemplatesImportExportTests(OrchestraTransactionTestCase):
+
+    def setUp(self):
+        super().setUp()
+        setup_models(self)
+        self.worker = Worker.objects.get(user__username='test_user_6')
+        self.worker.user.is_staff = True
+        self.worker.user.save()
+        self.request_client.login(username=self.worker.user.username,
+                                  password='defaultpassword')
+        self.full_todo_list_template = TodoListTemplateFactory(
+            slug='full-template-slug',
+            name='Full template name',
+            description='Full template description',
+            conditional_property_function={
```

Comment:
Good question! Our todo list templates are dynamic, meaning the machine can skip (mark irrelevant) or remove (outright delete) entire subtrees. The conditional property function returns a dictionary of facts about a project. The `skip_if` and `remove_if` entries of any node in the `todos` tree contain conjunctive normal form (ANDs of logical operations) that operate on the facts returned from conditional property function to determine whether to skip or remove their part of the tree.

That answers what the `condition_property_function` is. We don't have to import/export it because we're only import/exporting the `todos` tree (and logical operators), not the fact-returning function.

(It goes without saying that updating the Orchestra docs on this stuff would be a huge service.)

Reply:
Gotcha. Another nice task for the future might be to validate on import that the `skip_if` and `remove_if` logic only references valid fields output by the `conditional_property_function` somehow.